### ISSUE TITLE
[Bug] `jsx-no-constructed-context-values` empty schema is incorrect

### ISSUE THREAD

**Issue by Dimava** [jsx-eslint/eslint-plugin-react#3580]
https://github.com/jsx-eslint/eslint-plugin-react/blob/747fad0e06e73155616df62575626f7953260dec/lib/rules/jsx-no-constructed-context-values.js#LL141C13-L141C15

`jsx-no-constructed-context-values` empty schema should be `[]`, but it's `{}`

This leads to incorrect rule type generation by https://github.com/Shinigami92/eslint-define-config

```ts
export interface JsxNoConstructedContextValuesRule {
  /**
   * Disallows JSX context provider values from taking values that will cause needless rerenders.
   *
   * @see [jsx-no-constructed-context-values](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/jsx-no-constructed-context-values.md)
   */
  'react/jsx-no-constructed-context-values': Rule<
    [
      RuleLevel,
      ...{
        [k: string]: any; // <-- `...{}`, explected `...[]` (i.e. no settings)
      },
    ]
  >;
}
```

**Comment by ljharb** [comment]
I'm not familiar with that tool, but #3106 fixes that - it'd be a breaking change tho, so it's unlikely we'll ever do it.
